File: LL/src/Presentation/ll/src/app/shared/models/Dtos/market-place/market-place-buy-order.ts
import { ItemBase } from '../../item';

export interface MarketPlaceBuyOrder {
  id: string;
  buyerId: string;
  buyerName: string;
  itemBaseId: string;
  itemBase: ItemBase;
  quantity: number;
  unitPrice: number;
  createdAt: Date;
}

File: LL/src/Core/Domain/Models/MarketPlaces/MarketPlaceBuyOrder.cs
using Domain.Models.Items;

namespace Domain.Models.MarketPlaces;

public class MarketPlaceBuyOrder
{
    public Guid Id { get; set; }
    public Guid BuyerId { get; set; }
    public string BuyerName { get; set; } = string.Empty;
    public string ItemBaseId { get; set; } = string.Empty;
    public ItemBase ItemBase { get; set; } = null!;
    public int Quantity { get; set; }
    public long UnitPrice { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset ExpiresAt { get; set; }
}
